### PR TITLE
add stake/unstake call for pallet-stake v2

### PR DESCRIPTION
**Pull request by yz89**

### CODE REVIEW

File: pallets/liquid-staking-v2/src/lib.rs [hunk 1]
```diff
@@ -136,11 +282,38 @@ mod pallet {
...
     #[pallet::storage]
     #[pallet::getter(fn staking_pool)]
     pub type StakingPool<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
 
+    /// Records stake detail during each era.
+    #[pallet::storage]
+    #[pallet::getter(fn account_stake)]
+    pub type AccountStake<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
```
Comment (GopherJ) [inline pallets/liquid-staking-v2/src/lib.rs:295]: we should maybe use blake hash  for era index & account id

File: pallets/liquid-staking-v2/src/lib.rs [hunk 2]
```diff
@@ -136,11 +282,38 @@ mod pallet {
...
+        ValueQuery,
+    >;
+
+    /// Record all the pending unstaking requests.
+    /// Key is the owner of assets.
+    #[pallet::storage]
+    #[pallet::getter(fn account_unstake)]
+    pub type AccountUnstake<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
```
Comment (GopherJ) [inline pallets/liquid-staking-v2/src/lib.rs:309]: same as above

File: pallets/liquid-staking-v2/src/lib.rs [hunk 3]
```diff
@@ -92,8 +106,140 @@ mod pallet {
...
+        /// further used as collateral for lending.
+        ///
+        /// - `amount`: the amount of staking assets
+        #[pallet::weight(T::WeightInfo::stake())]
+        #[transactional]
+        pub fn stake(origin: OriginFor<T>, amount: Balance) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            let exchange_rate = ExchangeRate::<T>::get();
+            let derivative_amount = exchange_rate
```
Comment (GopherJ) [inline pallets/liquid-staking-v2/src/lib.rs:163]: I think we are mixing two words here:

liquid & derivative

maybe we can choose one and keep it consistent

**Comment by GopherJ** [comment]
I think you can create branch on parallel repo so that the others can also modify and help editing

**Comment by GopherJ** [comment]
the `TotalStake` per era & `TotalUnstake` may need to be recorded as well

File: pallets/liquid-staking-v2/src/lib.rs (hunk 1 above)
Comment (yz89) [reply pallets/liquid-staking-v2/src/lib.rs:295]: addressed by using Blake2_128Concat for EraIndex.

File: pallets/liquid-staking-v2/src/lib.rs (hunk 2 above)
Comment (yz89) [reply pallets/liquid-staking-v2/src/lib.rs:309]: addressed by using Blake2_128Concat for EraIndex.

File: pallets/liquid-staking-v2/src/lib.rs (hunk 3 above)
Comment (yz89) [reply pallets/liquid-staking-v2/src/lib.rs:163]: changed to liquid_amount

**Comment by yz89** [comment]
> the `TotalStake` per era & `TotalUnstake` may need to be recorded as well

yes, I added it in the next commit.

**Comment by yz89** [comment]
> I think you can create branch on parallel repo so that the others can also modify and help editing

OK, I will close this PR and use a parallel/branch.

**Pull request closed without merging**